pkg/manifests/manifests.go

```diff
@@ -2098,6 +2112,89 @@ func (f *Factory) PrometheusAdapterAPIService() (*apiregistrationv1.APIService,
 	return f.NewAPIService(f.assets.MustNewAssetReader(PrometheusAdapterAPIService))
 }
 
+func (f *Factory) MetricsServerServiceAccount() (*v1.ServiceAccount, error) {
+	return f.NewServiceAccount(f.assets.MustNewAssetReader(MetricsServerServiceAccount))
+}
+
+func (f *Factory) MetricsServerClusterRole() (*rbacv1.ClusterRole, error) {
+	return f.NewClusterRole(f.assets.MustNewAssetReader(MetricsServerClusterRole))
+}
+
+func (f *Factory) MetricsServerClusterRoleBinding() (*rbacv1.ClusterRoleBinding, error) {
+	return f.NewClusterRoleBinding(f.assets.MustNewAssetReader(MetricsServerClusterRoleBinding))
+}
+
+func (f *Factory) MetricsServerClusterRoleAggregatedMetricsReader() (*rbacv1.ClusterRole, error) {
+	return f.NewClusterRole(f.assets.MustNewAssetReader(MetricsServerClusterRoleAggregatedMetricsReader))
+}
+
+func (f *Factory) MetricsServerClusterRoleBindingAuthDelegator() (*rbacv1.ClusterRoleBinding, error) {
+	return f.NewClusterRoleBinding(f.assets.MustNewAssetReader(MetricsServerClusterRoleBindingAuthDelegator))
+}
+
+func (f *Factory) MetricsServerRoleBindingAuthReader() (*rbacv1.RoleBinding, error) {
+	return f.NewRoleBinding(f.assets.MustNewAssetReader(MetricsServerRoleBindingAuthReader))
+}
+
+func (f *Factory) MetricsServerDeployment() (*appsv1.Deployment, error) {
+	dep, err := f.NewDeployment(f.assets.MustNewAssetReader(MetricsServerDeployment))
+	if err != nil {
+		return nil, err
+	}
+
+	podSpec := &dep.Spec.Template.Spec
+	containers := podSpec.Containers
+	idx := slices.IndexFunc(containers, containerNameEquals("metrics-server"))
+	if idx < 0 {
+		return nil, errors.Errorf(
+			"failed to find metrics-server container %q in deployment %q",
+			"metrics-server", MetricsServerDeployment)
+	}
+
+	containers[idx].Image = f.config.Images.MetricsServer
+	containers[idx].Args = f.setTLSSecurityConfiguration(podSpec.Containers[0].Args,
+		MetricsServerTLSCipherSuitesFlag, MetricsServerTLSMinTLSVersionFlag)
+
+	config := f.config.ClusterMonitoringConfiguration.MetricsServerConfig
+	if config == nil {
+		return dep, nil
+	}
+
+	if len(config.NodeSelector) > 0 {
+		podSpec.NodeSelector = config.NodeSelector
```

Comment:
q: For all the config fields if not specified do we want to copy the value from prometheus-adapter deployment?

Reply:
eventually yes but it can/should be a follow-up PR.